positional_encodings/tf_positional_encodings.py

```diff
@@ -1,13 +1,26 @@
 import tensorflow as tf
 import numpy as np
 
-
 class TFPositionalEncoding2D(tf.keras.layers.Layer):
-    def __init__(self, channels, dtype=tf.float32):
+    def __init__(self, channels:int, return_format:str="sum", dtype=tf.float32):
         """
-        :param channels: The last dimension of the tensor you want to apply pos emb to.
+        Args:
+            channels int: The last dimension of the tensor you want to apply pos emb to.
+
+        Keyword Args:
+            return_format str: Return either the position encoding "pos" or the sum
+                of the inputs with the position encoding "sum". Default is "sum".
+            dtype: output type of the encodings. Default is "tf.float32".
+
         """
         super(TFPositionalEncoding2D, self).__init__()
+        return_formats = {
+            "pos": 0,
+            "sum": 1
+        }
+        if return_format not in return_formats:
+            raise ValueError(f'"{return_format}" is an unkown return format. Value must be in {[key for key in return_formats]}')
```

Comment:
Is there a particular reason that the `return_formats` has to be a dictionary? We could make it a set instead, like so:

```python
return_formats = {"pos", "sum"}
```

Reply:
Sound fair, I was in enum mode from working in C# a lot. So I thought this might be the best next thing which is used in python. But sure, string comparing should work fine as well.